Services/Services/SourceManagerService.cs

```diff
@@ -15,9 +15,53 @@
 * If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
 */
 
+using Models;
+using Services.Sources;
+
 namespace Services.Services
 {
+    /// <summary>
+    /// SourceManagerService
+    /// </summary>
+    public class SourceManagerService
+    {
+        /// <summary>
+        /// The git lab source
+        /// </summary>
+        private readonly ISource gitLabSource;
+        /// <summary>
+        /// The git hub source
+        /// </summary>
+        private readonly ISource gitHubSource;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SourceManagerService"/> class.
+        /// </summary>
+        /// <param name="gitlabSource">The gitlab source.</param>
+        /// <param name="githubSource">The github source.</param>
+        public SourceManagerService(GitLabSource gitlabSource, GitHubSource githubSource)
+        {
+            this.gitLabSource = gitlabSource;
+            this.gitHubSource = githubSource;
 
-    public class SourceManagerService { }
+        }
+        /// <summary>
+        /// Fetches the project.
+        /// </summary>
+        /// <param name="repoUrl">The repo URL.</param>
+        /// <returns></returns>
+        public Project FetchProject(string repoUrl)
```

Comment:
The GitHub source seems to be missing. Yet this PR is still linked to and closes the issues of both GitLab and GitHub. Did you forget the WIP tag?

In the future, other sources will be added to this function. These sources will not only be URL's but also other types of URI's. The project resource (in our database) and the front end wizard already support URI's. Please keep this in mind while developing this feature (At the moment it says URL everywhere, while this should be URI).

Reply:
See this explenation about why url
https://danielmiessler.com/study/difference-between-uri-url/